### PR TITLE
Send signal whenever topmost window id changes.

### PR DESCRIPTION
**Pull request by jusa**

### CODE REVIEW

File: src/compositor/lipstickcompositor.cpp [hunk 1]
```diff
@@ -17,6 +17,7 @@
 #include <contentaction.h>
 #endif
 
+#include <QtDBus>
```
Comment (SfietKonstantin) [inline src/compositor/lipstickcompositor.cpp]: Including everything in DBus might slow compilation a bit.

**Comment by SfietKonstantin** [comment]
I wonder what would happen if we have several "toplevel" windows (like split view)

**Comment by spiiroin** [comment]
Maybe array of toplevel windows could be used to allow supporting split view later on

**Comment by jusa** [comment]
@spiiroin @SfietKonstantin Hmm. Sending array of toplevel windows could work. I have no idea how the split view would be implemented in lipstick, but an array sounds like something that could stay the same even with split view implementation.

**Comment by spiiroin** [comment]
Also, since signals are just changes -> there should be a method call for getting the current/initial state too.

**Comment by spiiroin** [comment]
Hmmm... if lipstick already has dbus properties, maybe this should be one too?

**Comment by mikkoharju** [comment]
If the property closely corresponds to the "focused" window ie. the one where the user expects any hw key input to be directed, then there should be only one value (or no value), not an array. For an array, the semantics would be closer to "visible top level windows" ? The purpose of the addition is to help "OHM to determine the media state". @jusa, is it more useful to have an array of pids for OHM, or just a single one? Also, calling it a "topmost" does not tell the return value is a PID. "activeProcess"? "currentProcess"? or maybe expose it like /org/nemomobile/lipstick/currentApplication/[pid, name, class] ?

**Comment by jusa** [comment]
@mikkoharju With "split view" it might be more useful to know all visible top level windows. In OHM the PIDs are mapped to applications that have requested audio playback resource.

The end result is supposed to be that we know whether application that wants to play audio is in foreground or not. I haven't decided yet whether it is better to consider all visible apps in split view scenario or only the active one. When we know that audio playing app is in foreground or visible, hw volume control would change media volume, and when no audio playing apps are visible or playing audio, hw volume control would change ringtone volume.

**Comment by mikkoharju** [comment]
We tried to come up with quick window model dbus api but I'd say better keep it simple first. So, a dbus property for topmost process id, with a property name that discourages using it would be good. We'd need to come up with other reasonable use cases for the full list to make it worth it.

**Comment by mikkoharju** [comment]
:+1: lgtm

**Pull request merged**